bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/TeaserImpl.java

```diff
@@ -57,6 +83,49 @@ public String getLinkURL() {
         return super.getLinkURL();
     }
 
+    @PostConstruct
+    protected void initModel() {
+        super.initModel();
+        ValueMap properties = resource.getValueMap();
+        titleFromPage = properties.get(Teaser.PN_TITLE_FROM_PAGE, titleFromPage);
+        descriptionFromPage = properties.get(Teaser.PN_DESCRIPTION_FROM_PAGE, descriptionFromPage);
+    }
+
+    @Override
+    public String getTitle() {
+        if (this.title == null && !this.titleHidden) {
+            if (titleFromPage) {
+                this.title = this.getTargetPage()
+                        .map(tp -> StringUtils.defaultIfEmpty(tp.getPageTitle(), tp.getTitle()))
+                        .orElseGet(() -> this.getTeaserActions().stream().findFirst()
+                                .map(com.adobe.cq.wcm.core.components.internal.models.v1.TeaserImpl.Action::getTitle)
+                                .orElseGet(() -> Optional.ofNullable(getCurrentPage())
+                                        .map(cp -> StringUtils.defaultIfEmpty(cp.getPageTitle(), cp.getTitle()))
+                                        .orElse(null)));
+            } else {
+                this.title = this.resource.getValueMap().get(JcrConstants.JCR_TITLE, String.class);
+            }
+        }
+        return title;
+    }
+
+    @Override
+    public String getDescription() {
+        if (this.description == null && !this.descriptionHidden) {
+            if (descriptionFromPage) {
+                this.description = this.getTargetPage().map(Optional::of).orElseGet(() -> Optional.ofNullable(getCurrentPage()))
```

Comment:
the logic here should be the same as for the title: in case there is no target page, the description should be the one of the first CTA. If no CTA is defined, it should take the description of the current page.

Reply:
Please make sure the logic is the same as for the link:
```
        link = this.getTargetPage()
                .map(page -> Optional.of(linkHandler.getLink(page.getPath(), linkTarget).orElse(null)))
                .orElseGet(() -> {
                    // target page doesn't exist
                    if (this.isActionsEnabled()) {
                        return this.getActions().stream().findFirst()
                                .map(action -> Optional.ofNullable(linkHandler.getLink(action.getURL(), null).orElse(null)))
                                .orElse(Optional.empty());
                    } else {
                        // use the property value if actions are not enabled
                        return Optional.ofNullable(linkHandler.getLink(resource, Link.PN_LINK_URL).orElse(null));
                    }
                });

```

Reply:
@jckautzmann , `getTargetPage` already tries to get the first page from the CTAs. For the link we revisit the CTAs/actions because there might be an external/non-page link.